crates/ty_python_semantic/src/program.rs

```diff
@@ -37,8 +38,10 @@ impl Program {
         let search_paths = SearchPaths::from_settings(db, &search_paths)
             .with_context(|| "Invalid search path settings")?;
 
-        let python_version_with_source =
-            Self::resolve_python_version(python_version_with_source, search_paths.python_version());
+        let python_version_with_source = Self::resolve_python_version(
+            python_version_with_source,
+            search_paths.python_version().as_deref(),
```

Comment:
Should we pass the `search_paths` to `resolve_python_version` instead to avoid calling `to_python_version` when `python_version_with_source` isn't `None`?